Enable default sitich output parameters to be pulled from IPF

sf1919 opened the pull request:
**Description of work.**
The first step in completing #34843 has been to put in the functionality to allow default stitch parameters to be taken from IPF files and added to the Experiment Tab on the Reflectometry GUI. A separate PR will be follow with changes to POLREF IPF and will include the release note.


**To test:**
1) Testing GUI still works/default is empty
- Open Mantid
- Open ISIS Reflectometry GUI
- Go to Experiment Settings Tab and check the `Output Stitch Properties` box at the bottom of the GUI is empty

2) Test it will pull in defaults when they exist
- Edit the `INTER_Parameters.xml` file that Mantid will use. This will either be located in your development folders (`../instrument/INTER_Parameters.xml`) or for Windows users it might be in `../Users/yourFedID/AppData/Roaming/mantidproject/mantid/instrument/INTER_Parameters.xml` . Add the following bit of code 

```  xml
<parameter name="StitchManualScaleFactors" type="string">
    <value val="1"/>
</parameter> 
```
- Open Mantid
- Open ISIS Reflectometry GUI
- Go to Experiment Settings Tab and check the `Output Stitch Properties` box at the bottom of the GUI should have `ManualScaleFactors='1'` in it
- Don't forget to remove edits from your IPF file when you're finished!


*This does not require release notes* because **they will be added in the 2nd PR for this issue**.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Enable Output Stitch Parameters to be taken from IPF

- The ExperimentOptionDefaults now looks up default properties
that can be used in the Stitch1DMany algorithm. These are then added
to the Output Stitch Parameters field of the Experiment Options tab
in the Reflectometry GUI.

sf1919 committed:
Improve OptionDefaults method

- allows for any algorithm to be used with OptionDefaults

sf1919 committed:
Changed how map of Stitch parameters created

sf1919 committed:
Add null and parameter tests

ConorMFinn commented on qt/scientific_interfaces/ISISReflectometry/test/Experiment/ExperimentOptionDefaultsTest.h:
```diff
@@ -141,6 +141,16 @@ class ExperimentOptionDefaultsTest : public CxxTest::TestSuite {
+
+  void testValidStitchParamsOptionsFromParamsFile() {
+    auto result = getDefaultsFromParamsFile("Experiment");
+    TS_ASSERT_EQUALS(result.stitchParameters().size(), 1);
```
Could we add a check that the right name and value have been taken from the IPF?

cailafinn reviewed (changes requested):
Works well, it's nice to have this added. Just a few comments on the code:

sf1919 committed:
Changes made after review

- added const values where appropriate
- refactored code to make it simpler
- expanded test to check values in map are the ones expected

cailafinn reviewed (changes requested):
Just one more thing on variable naming, but looks really good now 👍

sf1919 committed:
Make variable names more relevant

ConorMFinn commented on qt/scientific_interfaces/ISISReflectometry/GUI/Experiment/ExperimentOptionDefaults.cpp:
```diff
@@ -12,6 +12,7 @@
 #include "LookupTableValidator.h"
 #include "MantidAPI/AlgorithmManager.h"
 #include "Reduction/Experiment.h"
+#include "Reduction/ParseReflectometryStrings.h"
```
I don't think you're using this anymore

cailafinn reviewed (changes requested):
Just one more unused `#include` that I noticed. Otherwise all looks good!

sf1919 committed:
Remove unused include

cailafinn reviewed (approved):
Looks good to me, nice to have this in! 👍 Nice work!
